### PR TITLE
Reduce the peak memory usage of skillmodels

### PR DESCRIPTION
**Pull request by hmgaudecker**
It looks like memory consumption of `loglike_and_grad` is rising about linearly in the number of observations (and probably also parameters) at present, making it the limiting factor for real-world applications. This PR aims at reducing peak memory usage.

Table to keep track of mem usage of `pixi run mem-on-clean-repo` will be in MEMTRACKER.md. Delete upon merging.

### CODE REVIEW

File: tests/test_kalman_filters.py [hunk 1]
```diff
@@ -238,30 +232,25 @@ def test_predict_against_linear_filterpy(seed):
     expected_cov = fp_filter.P
 
     def linear(params, states):
-        return np.dot(states, params)
```
Comment (janosg) [inline tests/test_kalman_filters.py:241]: Wow! You should definitely check performance on GPU again after this fix.

File: src/skillmodels/kalman_filters.py [hunk 2]
```diff
@@ -177,9 +182,6 @@ def kalman_predict(
             the sigma_point algorithm chosen.
         sigma_weights (jax.numpy.array): 1d array of length n_sigma with non-negative
             sigma weights.
-        transition_info (dict): Dict with the entries "func" (the actual transition
```
Comment (janosg) [inline src/skillmodels/kalman_filters.py:180]: Was the docstring of `transition_info` outdated or did you actually change this? If you changed it, why is the mapping from columns to positions no longer needed?

Comment (hmgaudecker) [reply src/skillmodels/kalman_filters.py:180]: I did not change any functionality, but did check that the only thing ever used was the function. No idea when this stopped being required, for sure had been the case [2.5 years ago](https://github.com/OpenSourceEconomics/skillmodels/commit/22fb915a894ba12dd8a90822680ec380dcadc1a4#diff-f67fef4e2f70a1271df3182635d6e146f32efbe1518e83b7211f36d0d3422975).

Comment (hmgaudecker) [reply src/skillmodels/kalman_filters.py:180]: *(so yes, I did actually change from passing a dict to passing the function only)*

Comment (janosg) [reply src/skillmodels/kalman_filters.py:180]: Maybe this was before we went full dags? But if it wasn't used you can of course remove it.

**Review by janosg — APPROVED** [review]

**Comment by hmgaudecker** [comment]
For the record: The 8 lines of missed coverage are `_get_jnp_params_vec`, which does not have tests. Does not seem to be an issue.

**Pull request merged**